registry/paging_test.go

```diff
@@ -0,0 +1,142 @@
+package registry
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+func TestPages(t *testing.T) {
+	http.HandleFunc("/v2/", pingHandler)
+	http.HandleFunc("/v2/example.com/image/tags/list", tagsHandler)
+	http.HandleFunc("/v2/_catalog", catalogHandler)
+	go http.ListenAndServe(":9999", nil)
+
+	hub, err := New("http://localhost:9999/", "", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ts, err := hub.Tags("example.com/image")
+	checkItems(t, tags, ts)
+
+	rs, err := hub.Repositories()
+	checkItems(t, repos, rs)
+}
+
+const perPage = 5
+
+var tags = []string{
+	"tag1",
+	"tag2",
+	"tag3",
+	"tag4",
+	"tag5",
+	"tag6",
+	"tag7",
+	"tag8",
+	"tag9",
+	"tag10",
+	"tag11",
+	"tag12",
+}
+
+var repos = []string{
+	"image1",
+	"image2",
+	"image3",
+	"image4",
+	"image5",
+	"image6",
+	"image7",
+	"image8",
+	"image9",
+	"image10",
+	"image11",
+	"image12",
+	"image13",
+	"image14",
+	"image15",
+	"image16",
+	"image17",
+}
+
+func checkItems(t *testing.T, expected []string, got []string) {
```

Comment:
You can use [reflect.DeepEqual](https://golang.org/pkg/reflect/#DeepEqual) here.

```
if !reflect.DeepEqual(expected, got) {
  t.Fatalf("Cannot match entries. Expected %q, got %q", expected, got)
}
```

With this simplification, I don't think this method is necessary anymore. Everything can go into the test.